ionoscloud/resource_dbaas_mariadb_cluster_test.go

```diff
@@ -0,0 +1,225 @@
+			return fmt.Errorf("an error occured while fetching MariaDB cluster with ID: %v, error: %w", rs.Primary.ID, err)
+		}
+		if *foundCluster.Id != rs.Primary.ID {
+			return fmt.Errorf("resource not found")
+		}
+		cluster = &foundCluster
+
+		return nil
+	}
+}
+
+const mariaDBClusterConfigBasic = `
+resource ` + constant.DatacenterResource + ` ` + datacenterResourceName + ` {
+  name        = "mariadb_datacenter_example"
+  location    = "de/txl"
+  description = "Datacenter for testing MariaDB cluster"
+}
+
+resource ` + constant.LanResource + ` ` + lanResourceName + ` {
+  datacenter_id = ` + constant.DatacenterResource + `.datacenter_example.id 
+  public        = false
+  name          = "mariadb_lan_example"
+}
+
+resource ` + constant.DBaaSMariaDBClusterResource + ` ` + constant.DBaaSClusterTestResource + ` {
+  ` + clusterVersionAttribute + ` = "` + clusterVersionValue + `"
+  ` + clusterInstancesAttribute + ` = "` + clusterInstancesValue + `"
+  ` + clusterCoresAttribute + ` = "` + clusterCoresValue + `"
+  ` + clusterRamAttribute + ` = "` + clusterRamValue + `"
+  ` + clusterStorageSizeAttribute + ` = "` + clusterStorageSizeValue + `"
+  ` + clusterDisplayNameAttribute + ` = "` + clusterDisplayNameValue + `"
+  ` + connections + `
+  ` + maintenanceWindow + `
+  ` + credentials + `
+}
+
+resource ` + constant.RandomPassword + ` "cluster_password" {
+  length           = 16
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+`
+
+const mariaDBClusterDataSourceMatchId = mariaDBClusterConfigBasic + `
+data ` + constant.DBaaSMariaDBClusterResource + ` ` + constant.DBaaSClusterTestDataSourceById + ` {
+	id = ` + constant.DBaaSMariaDBClusterResource + `.` + constant.DBaaSClusterTestResource + `.id
+}
+`
+
+const mariaDBClusterDataSourceMatchName = mariaDBClusterConfigBasic + `
+data ` + constant.DBaaSMariaDBClusterResource + ` ` + constant.DBaaSClusterTestDataSourceByName + ` {
+	display_name	= "` + clusterDisplayNameValue + `"
+}
+`
+
+const mariaDBClusterDataSourceWrongName = `
+data ` + constant.DBaaSMariaDBClusterResource + ` ` + constant.DBaaSClusterTestDataSourceByName + ` {
+  display_name = "wrong_name"
+}
+`
+
+// Any valid UUID can be used here since there is no cluster created, so no cluster will be found
+const mariaDBClusterDataSourceWrongId = `
+data ` + constant.DBaaSMariaDBClusterResource + ` ` + constant.DBaaSClusterTestDataSourceById + ` {
+  id = "178d6a7d-5ed4-44de-88f0-27f1182d8ae8"
+}
+`
+
+// Attributes
+const clusterVersionAttribute = "mariadb_version"
+
+// Values
+const clusterVersionValue = "10.6"
+const clusterInstancesValue = "1"
+const clusterCoresValue = "4"
+const clusterRamValue = "2"
+const clusterStorageSizeValue = "10"
+const clusterConnectionsCidrValue = "192.168.1.100/24"
+const clusterDisplayNameValue = constant.DBaaSClusterTestResource
+const clusterCredentialsUsernameValue = "username"
+const datacenterResourceName = "datacenter_example"
+const lanResourceName = "lan_example"
```

Comment:
we can group these in a const block, and maybe move them at the top or bottom
that way we group single-values separately from hcl blocks

Reply:
@cbeti-ionos done in eabe0a8e1529e0c3b0c3d8b882cc4b2d6056674f